californium-core/src/main/java/org/eclipse/californium/core/network/serialization/DataParser.java

```diff
@@ -53,17 +55,22 @@ public final Message parseMessage(final RawData raw) {
 	 */
 	public final Message parseMessage(final byte[] msg) {
 
+		String message = "illegal message code";
 		DatagramReader reader = new DatagramReader(msg);
 		MessageHeader header = parseHeader(reader);
-		if (CoAP.isRequest(header.getCode())) {
-			return parseMessage(reader, header, new Request(CoAP.Code.valueOf(header.getCode())));
-		} else if (CoAP.isResponse(header.getCode())) {
-			return parseMessage(reader, header, new Response(CoAP.ResponseCode.valueOf(header.getCode())));
-		} else if (CoAP.isEmptyMessage(header.getCode())) {
-			return parseMessage(reader, header, new EmptyMessage(header.getType()));
-		} else {
-			throw new MessageFormatException("illegal message code", header.getMID(), header.getCode(), CoAP.Type.CON == header.getType());
+		try {
+			if (CoAP.isRequest(header.getCode())) {
+				return parseMessage(reader, header, new Request(CoAP.Code.valueOf(header.getCode())));
+			} else if (CoAP.isResponse(header.getCode())) {
+				return parseMessage(reader, header, new Response(CoAP.ResponseCode.valueOf(header.getCode())));
+			} else if (CoAP.isEmptyMessage(header.getCode())) {
+				return parseMessage(reader, header, new EmptyMessage(header.getType()));
+			}
+		} catch (MessageFormatException e) {
```

Comment:
MessageFormatException are throw by util methods, which don't have the CoAP message context (MID, Code, CON/NON, e.g. CoAP.Code.valueOf). Trying to test the changes in "reject(final RawData raw, final MessageFormatException cause) " #179, I recognized that.
So catching and filling the context would allow to process such messages as CoAP message in `receiveMessage(final RawData raw)`. And as mentioned above, this increase the number of RST instead of ignored messages.